### PR TITLE
Improve readability by packing frequently used variables into namedtuples.

### LINKED ISSUE #67: `ProposalStateType`, `IntegratorStateType`, etc should be `namedtuple`s
Even if we can only pass tensors in functions used in `scan` this will make this code a lot easier to follow, and ready in case it becomes possible to pass namedtuples in `scan`.

__Originally posted by @brandonwillard in https://github.com/aesara-devs/aehmc/pull/7#discussion_r678733304__

### PR DESCRIPTION
**Pull request by zoj613**
closes #67 

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [ ] There is an informative high-level description of the changes.
+ [x] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [x] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [ ] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.  If your commit description starts with "Fix...", then you're probably making this mistake.
+ [ ] There are tests covering the changes introduced in the PR.

Don't worry, your PR doesn't need to be in perfect order to submit it.  As development progresses and/or reviewers request changes, you can always [rewrite the history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) of your feature/PR branches.

If your PR is an ongoing effort and you would like to involve us in the process, simply make it a [draft PR](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

### CODE REVIEW

**Comment by zoj613** [comment]
@rlouf could you please help diagnose the one failed test `test_nuts_mcse`? It appears that the MCSE fails for one of the cases in the NUTS sampler. The changes seem fine and do not appear to change behavior so i'm a little confused.

File: aehmc/window_adaptation.py [hunk 1]
```diff
@@ -54,12 +62,16 @@ def one_step(
...
         return (
             chain_state[0],  # q
             chain_state[1],  # potential_energy
             chain_state[2],  # potential_energy_grad
-            *warmup_state[0],
+            da_state.step,
+            da_state.iterates,  # log_step_size
+            da_state.iterates_avg,  # log_step_size_avg
+            da_state.gradient_avg,
+            da_state.shrinkage_pts,  # mu
```
Comment (brandonwillard) [inline aehmc/window_adaptation.py:74]: What are these comments indicating?

**Review by brandonwillard — DISMISSED** [review]
These changes look great!

File: aehmc/window_adaptation.py (hunk 1 above)
Comment (zoj613) [reply aehmc/window_adaptation.py:74]: The aual averaging state's "iterates" is represented by the `log_step_size` variable and the "shrinkage_pts" is represented by the `mu` variable in the context of this function. Is this correct, @rlouf ?

**Review by brandonwillard — DISMISSED** [review]

**Review by brandonwillard — APPROVED** [review]

**Comment by brandonwillard** [comment]
These updates look great; I'm ready to move forward with them.

**Comment by zoj613** [comment]
> These updates look great; I'm ready to move forward with them.

I will still need to look into updating the docstrings. @rlouf any thoughts on these changes?

**Comment by brandonwillard** [comment]
Merged, because these changes are already very helpful.  We can add follow-ups for the docstrings and anything else.

**Pull request merged**